internal/reader/ccip.go

```diff
@@ -216,13 +210,79 @@ func (r *CCIPChainReader) GasPrices(ctx context.Context, chains []cciptypes.Chai
 }
 
 func (r *CCIPChainReader) Sync(ctx context.Context) (bool, error) {
-	return false, nil
+	sourceConfigs, err := r.getSourceChainsConfig(ctx)
+	if err != nil {
+		return false, fmt.Errorf("get onramps: %w", err)
+	}
+
+	for chain, cfg := range sourceConfigs {
+		if cfg.OnRamp == "" {
+			return false, fmt.Errorf("onRamp address not found for chain %d", chain)
+		}
+
+		// Bind the onRamp contract address to the reader.
+		// If the same address exists -> no-op
+		// If the address is changed -> updates the address, overwrites the existing one
+		// If the contract not binded -> binds to the new address
+		if err := r.contractReaders[chain].Bind(ctx, []types.BoundContract{
```

Comment:
Can we check if it's an update and log? Can be helpful for debugging purposes.

Reply:
no we can't unfortunately